routes/SpotsRouter.js

```diff
@@ -0,0 +1,62 @@
+import express from "express";
+import moment from "moment";
+import {ParkSpace} from "../models/ParkSpace.js";
+import {BOOKING_STATUS} from "../utils/BookingStatus.js";
+import Booking from "../models/Booking.js";
+
+const totalSpotsRouter = express.Router();
+
+totalSpotsRouter.post("/isSpotAvailable", async (req, res) => {
+    try {
+        const reqCheckIn = moment(req.body.check_in).format()
+        const reqCheckOut = moment(req.body.check_out).format()
+
+        let totalSpots = {}
+
+        const pendingAndInuseBookings = await Booking.find({$or: [{booking_status: BOOKING_STATUS.PENDING}, {booking_status: BOOKING_STATUS.INUSE}]})
+        const fulfilledAndCancelledBookings = await Booking.find({$or: [{booking_status: BOOKING_STATUS.FULFILLED}, {booking_status: BOOKING_STATUS.CANCELLED}]})
+
+        let availableSpots = fulfilledAndCancelledBookings.map(getAvailableSpaceId)
+        let unavailableSpots = pendingAndInuseBookings.map(getUnavailableSpaceId)
+
+        let unavailableParkSpaces = await ParkSpace.find({_id: {$in: unavailableSpots}}, {space_name: 1, _id: false});
+        let availableParkSpaces = await ParkSpace.find({_id: {$in: availableSpots}}, {space_name: 1, _id: false});
+
+        function getAvailableSpaceId(booking) {
+            let spaces = []
+            spaces.push(booking.space_id)
+            return spaces
+        }
+
+        function getUnavailableSpaceId(booking) {
```

Comment:
This won't work. You pushing  to an array while looping through the pendingAndInUseBookings, doesn't make sense.


Use the example below to fix it and make a callback for line 20
`const euro = [29.76, 41.85, 46.5];

const above30 = euro.reduce((total, amount) => {
  if (amount > 30) {
    total.push(amount);
  }
  return total;
}, []);

above30 // [ 41.85, 46.5 ]`